### PR TITLE
NATS Streaming: Add parallel processing, fix retry bug

### PR DESCRIPTION
**Pull request by yaron2**
Fixes https://github.com/dapr/components-contrib/issues/1669.

This PR adds parallel processing support to NATS streaming and fixes a bug where a single Dapr instance would continuously retry messages locally, keeping a failed message from being delivered to a different consumer until the current instance stopped running. The internal retry mechanism here has been removed per our ongoing effort to consolidate on the resiliency mechanism.

### CODE REVIEW

**Comment by yaron2** [comment]
@daixiang0 can you please review? thanks!

File: pubsub/natsstreaming/natsstreaming_test.go [hunk 1]
```diff
@@ -189,10 +193,24 @@ func TestParseNATSStreamingMetadataForValidSubscriptionOptions(t *testing.T) {
...
 				consumerID:             "consumer1",
 				subscriptionType:       "topic",
+				pubsub.ConcurrencyKey:  "single",
 				startAtTimeDelta:       "1h",
 			},
 			"startAtTimeDelta", "1h",
 		},
+
+		{
+			"using concurrencyMode",
```
Comment (daixiang0) [inline pubsub/natsstreaming/natsstreaming_test.go:203]: Please add a parallel concurrency test.

Comment (yaron2) [reply pubsub/natsstreaming/natsstreaming_test.go]: > Please add a parallel concurrency test.

Added, thanks.

**Review by daixiang0 — APPROVED** [review]
LGTM

**Pull request merged**